src/Menu/main.vala

```diff
@@ -0,0 +1,59 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1 --pkg pango --pkg glib-2.0 --pkg gio-2.0
+
+public void main () {
+
+    var label = (Gtk.Label) workbench.builder.get_object ("label");
+
+    var text_group = new SimpleActionGroup ();
+    label.insert_action_group ("text", text_group);
+
+    var text_state = new GLib.HashTable<string, GLib.Variant> (str_hash, str_equal);
+    text_state.insert ("italic", false);
+    text_state.insert ("bold", false);
+    text_state.insert ("foreground", "green");
+
+    var italic_action = new SimpleAction.stateful ("italic", null, new Variant.boolean (false));
+    italic_action.notify["state"].connect (() => {
+        text_state.replace ("italic", italic_action.state.get_boolean ());
+        label.attributes = (state_to_attr (text_state));
+    });
+    text_group.add_action (italic_action);
+
+    var bold_action = new SimpleAction.stateful ("bold", null, new Variant.boolean (false));
+    bold_action.notify["state"].connect (() => {
+        text_state.replace ("bold", bold_action.state.get_boolean ());
+        label.attributes = (state_to_attr (text_state));
+    });
+    text_group.add_action (bold_action);
+
+    var color_action = new SimpleAction.stateful ("color", new GLib.VariantType ("s"), new Variant.string ("green"));
+    color_action.notify["state"].connect (() => {
+        text_state.replace ("foreground", color_action.state.get_string ());
+        label.attributes = (state_to_attr (text_state));
+    });
+    text_group.add_action (color_action);
+}
+
+// Helper function to create a PangoAttrList from text_state
+private Pango.AttrList state_to_attr (HashTable<string, Variant> state) {
+    string attr_string = "";
+    var attrs = new Gtk.StringList (null);
```

Comment:
Is there any reason to use `Gtk.StringList`? Otherwise a `GenericArray<string>` or just a string array would fit better.

Reply:
I followed the Python implementation and the code used `.append` method, no other reason :). Switched to `GenericArray<string>` and used `foreach`